libcloud/common/dimensiondata.py

```diff
@@ -296,20 +302,100 @@ def __repr__(self):
                    self.status))
 
 
+class DimensionDataPublicIpBlock(object):
+    """
+    DimensionData Public IP Block with location.
+    """
+
+    def __init__(self, id, baseIp, size, location, network_domain,
+                 status):
+        self.id = str(id)
+        self.baseIp = baseIp
+        self.size = size
+        self.location = location
+        self.network_domain = network_domain
+        self.status = status
+
+    def __repr__(self):
+        return (('<DimensionDataNetworkDomain: id=%s, baseIp=%s,'
+                 'size=%s, location=%s, status=%s>')
+                % (self.id, self.baseIp, self.size, self.location,
+                   self.status))
+
+
+class DimensionDataFirewallRule(object):
+    """
+    DimensionData Firewall Rule for a network domain
+    """
+
+    def __init__(self, id, name, action, location, network_domain,
+                 status, ipVersion, protocol, source, destination,
+                 enabled):
+        self.id = str(id)
+        self.name = name
+        self.action = action
+        self.location = location
+        self.network_domain = network_domain
+        self.status = status
+        self.ipVersion = ipVersion
+        self.protocol = protocol
+        self.source = source
+        self.destination = destination
+        self.enabled = enabled
+
+    def __repr__(self):
+        return (('<DimensionDataNetworkDomain: id=%s, name=%s,'
+                 'action=%s, location=%s, status=%s>')
+                % (self.id, self.name, self.action, self.location,
+                   self.status))
+
+
+class DimensionDataFirewallAddress(object):
+    """
+    The source or destination model in a firewall rule
+    """
+    def __init__(self, any_ip, ip_address, ip_prefix_size,
+                 port_begin, port_end):
+        self.any_ip = any_ip
+        self.ip_address = ip_address
+        self.ip_prefix_size = ip_prefix_size
+        self.port_begin = port_begin
+        self.port_end = port_end
+
+
+class DimensionDataNatRule(object):
+    """
+    An IP NAT rule in a network domain
+    """
+    def __init__(self, id, network_domain, internal_ip, external_ip, status):
+        self.id = id
+        self.network_domain = network_domain
+        self.internal_ip = internal_ip
+        self.external_ip = external_ip
+        self.status = status
+
+    def __repr__(self):
+        return (('<DimensionDataNatRule: id=%s, status=%s>')
+                % (self.id, self.status))
+
+
 class DimensionDataVlan(object):
     """
     DimensionData VLAN.
     """
 
-    def __init__(self, id, name, description, location, status):
+    def __init__(self, id, name, description, location, status,
+                 private_ipv4_range_address, private_ipv4_range_size):
```

Comment:
Is `private_ipv4_range_size` CIDR suffix (e.g. "32" part of ip/32 notation) or something else?